JASP-Desktop/widgets/filemenu/filemenu.cpp

```diff
@@ -411,39 +411,33 @@ bool FileMenu::checkSyncFileExists(const QString &path, bool waitForExistence)
 	if (path.startsWith("http"))
 		return true;
 
-	bool exist = false;
-	int maxExistTrials = waitForExistence ? 5 : 1;
-	int existTrials = 1;
+	auto checkExists = [](const QString& path) { return QFileInfo::exists(path); };
+	auto checkNotEmpty = [](const QString& path) { return Utils::getFileSize(path.toStdString()) > 0; };
 
-	while (!QFileInfo::exists(path) && existTrials < maxExistTrials)
+	auto checkFn = [&](bool (*f)(const QString&))
```

Comment:
:)

Reply:
Ooh yeah, but to make this also c++ instead of C:
`auto checkFn = [&](std::function<bool>(const QString &))`
Should also work right?